pkg/download/consistent_hashing_test.go

```diff
@@ -310,6 +310,49 @@ func TestConsistentHashRetries(t *testing.T) {
 	assert.Equal(t, "3344761726165516", string(bytes))
 }
 
+func TestConsistentHashRetriesMissingHostname(t *testing.T) {
+	hostnames := make([]string, len(testFSes))
+	for i, fs := range testFSes {
+		ts := httptest.NewServer(http.FileServer(http.FS(fs)))
+		defer ts.Close()
+		url, err := url.Parse(ts.URL)
+		require.NoError(t, err)
+		hostnames[i] = url.Host
+	}
+	// we want to test that we never fall back to origin. So we set origin to be
+	// this canary file and ensure we don't get any data from it
+	origin := "https://weights.replicate.delivery/hello.txt"
+
+	// we deliberately "break" this cache host to make it as if its SRV record was missing
+	hostnames[0] = ""
+
+	opts := download.Options{
+		Client:               client.Options{},
```

Comment:
Purely responding to your comment about this test making an upstream request: it might be nice to allow passing in a `Transport` in `client.Options`. If specified, it would be used instead of [the transport that's constructed here](https://github.com/replicate/pget/blob/451703749bb0b6ae46b6aa89fcc0cd7c1577a288/pkg/client/client.go#L61-L70).

Then you can pass in `httpmock.DefaultTransport` and use `httpmock` to intercept and/or set expectations around upstream requests.

Reply:
Absolutely. All of the tests should have an origin that shows we missed and went to origin.

Reply:
added issue #156 specifically for this comment.

Reply:
Passing in a transport makes a lot of sense, great suggestion